internal/backends/postgresql/backend.go

```diff
@@ -126,16 +128,31 @@ func (b *backend) ListDatabases(ctx context.Context, params *backends.ListDataba
 		return nil, err
 	}
 
-	res := &backends.ListDatabasesResult{
-		Databases: make([]backends.DatabaseInfo, len(list)),
-	}
+	var res *backends.ListDatabasesResult
 
-	for i, dbName := range list {
-		res.Databases[i] = backends.DatabaseInfo{
-			Name: dbName,
+	if params != nil && len(params.Name) > 0 {
+		i, found := slices.BinarySearchFunc(list, params.Name, func(dbName, t string) int {
```

Comment:
Is it safe to assume that the name on the list will always ordered then?

Reply:
Yes. That's the primary need. and tests cover that.
if you look at the test , the order of creation of Collection names and even Database names is different than wht you expect. and it covers this aspect.